### PR TITLE
tests(devtools): support dbw smoke

### PR DESCRIPTION
**Pull request by adamraine**
Would be nice to have DT smoke coverage of https://github.com/GoogleChrome/lighthouse/pull/14465. Turns out it was pretty easy to get dbw working in the DT smoke runner.

### CODE REVIEW

File: cli/test/smokehouse/test-definitions/dobetterweb.js [hunk 1]
```diff
@@ -36,15 +36,6 @@ const imgB = {
...
 const expectations = {
-  networkRequests: {
-    // Number of network requests differs between Fraggle Rock and legacy modes because
-    // FR has fewer passes, preserve this check moving forward.
-    _fraggleRockOnly: true,
-
-    // 22 requests made for a single navigation.
-    // 6 extra requests made because stylesheets are evicted from the cache by the time DT opens.
-    length: 28,
-  },
```
Comment (adamraine) [inline cli/test/smokehouse/test-definitions/dobetterweb.js:47]: https://github.com/GoogleChrome/lighthouse/pull/14465#discussion_r1047519919

**Review by brendankenny — APPROVED** [review]
Nice! Did not remember dbw was in the exclusion list. We should look into those "need further investigation" ones at some point too.

File: core/scripts/pptr-run-devtools.js [hunk 2]
```diff
@@ -315,10 +322,14 @@ async function testUrlFromDevtools(url, options = {}) {
     const logs = [];
     await installConsoleListener(inspectorSession, logs);
 
+    page.on('dialog', dismissDialog);
```
[the remark this answers is not in the document]
Comment (adamraine) [reply core/scripts/pptr-run-devtools.js:325]: It is working in DevTools. In the setup for the test however, the dialog prevents `pptr-run-devtools.js` from starting the Lighthouse run. This PR just dismissed the dialog in the initial test setup so we can actually start the test.

The dialog dismiss listener is removed before the test starts, and then Lighthouse must dismiss the dialog itself when it reloads the page.

[the remark this answers is not in the document]
Comment (brendankenny) [reply core/scripts/pptr-run-devtools.js:325]: Ah, I missed that this was running only before the test. Probably could use a comment but at least blame will find this :)

**Pull request merged**